L1Trigger/L1CaloTrigger/plugins/Phase2L1CaloPFClusterEmulator.cc

```diff
@@ -0,0 +1,266 @@
+      }
+    }
+    if(k % 2 ==  0) etaoffset = 0;
+    else etaoffset = 17;
+    if(k > 1 && k % 2 ==  0) phioffset = phioffset+4;
+
+    GCTPfcluster_t tempPfclusters;
+    tempPfclusters = pfcluster(temporary, etaoffset, phioffset);  
+
+    for(int i=0; i<8; i++){
+      int gcteta = tempPfclusters.GCTpfclusters[i].eta;
+      int gctphi = tempPfclusters.GCTpfclusters[i].phi;
+      float towereta = realEta[gcteta][gctphi];
+      float towerphi = realPhi[gcteta][gctphi];
+      //std::cout << "Phase2L1CaloPFClusterEmulator: GCT Pfclusters energy " << tempPfclusters.GCTpfclusters[i].et
+      //          << " at ieta, iphi: (" << tempPfclusters.GCTpfclusters[i].eta << ", " << tempPfclusters.GCTpfclusters[i].phi << ")"
+      //          << " at eta, phi: (" << towereta << ", " << towerphi << ")" << std::endl;
+      l1tp2::CaloPFCluster l1CaloPFCluster;
+      l1CaloPFCluster.setClusterEt(tempPfclusters.GCTpfclusters[i].et);
+      l1CaloPFCluster.setClusterIEta(gcteta);
+      l1CaloPFCluster.setClusterIPhi(gctphi);
+      l1CaloPFCluster.setClusterEta(towereta);
+      l1CaloPFCluster.setClusterPhi(towerphi);
+      pfclusterCands->push_back(l1CaloPFCluster);
+    }
+  }
+
+  iEvent.put(std::move(pfclusterCands), "GCTPFCluster");
+    
+}
+
+// ------------ method called once each stream before processing any runs, lumis or events  ------------
+void Phase2L1CaloPFClusterEmulator::beginStream(edm::StreamID) {
+  // please remove this method if not needed
+}
+
+// ------------ method called once each stream after processing all runs, lumis and events  ------------
+void Phase2L1CaloPFClusterEmulator::endStream() {
+  // please remove this method if not needed
+}
+
+// ------------ method called when starting to processes a run  ------------
+/*
+void
+Phase2L1CaloPFClusterEmulator::beginRun(edm::Run const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a run  ------------
+/*
+void
+Phase2L1CaloPFClusterEmulator::endRun(edm::Run const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when starting to processes a luminosity block  ------------
+/*
+void
+Phase2L1CaloPFClusterEmulator::beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a luminosity block  ------------
+/*
+void
+Phase2L1CaloPFClusterEmulator::endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void Phase2L1CaloPFClusterEmulator::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
+  descriptions.addDefault(desc);
+}
```

Comment:
Please fill the descriptions. The module doesn't have any input though, so it may not be critical.